routing_common/bicycle_model.cpp

```diff
@@ -26,33 +26,34 @@ using namespace routing;
-    /// @see Russia_UseTrunk for Trunk weights.
-    {HighwayType::HighwayTrunk, InOutCitySpeedKMpH(SpeedKMpH(5.5, 18.0))},
-    {HighwayType::HighwayTrunkLink, InOutCitySpeedKMpH(SpeedKMpH(5.5, 18.0))},
-    {HighwayType::HighwayPrimary, InOutCitySpeedKMpH(SpeedKMpH(10.0, 18.0), SpeedKMpH(14.0, 18.0))},
-    {HighwayType::HighwayPrimaryLink, InOutCitySpeedKMpH(SpeedKMpH(10.0, 18.0), SpeedKMpH(14.0, 18.0))},
-    {HighwayType::HighwaySecondary, InOutCitySpeedKMpH(SpeedKMpH(15.0, 18.0), SpeedKMpH(20.0, 18.0))},
-    {HighwayType::HighwaySecondaryLink, InOutCitySpeedKMpH(SpeedKMpH(15.0, 18.0), SpeedKMpH(20.0, 18.0))},
-    {HighwayType::HighwayTertiary, InOutCitySpeedKMpH(SpeedKMpH(15.0, 18.0), SpeedKMpH(20.0, 18.0))},
-    {HighwayType::HighwayTertiaryLink, InOutCitySpeedKMpH(SpeedKMpH(15.0, 18.0), SpeedKMpH(20.0, 18.0))},
-    {HighwayType::HighwayUnclassified, InOutCitySpeedKMpH(SpeedKMpH(12.0, 18.0))},
-
-    // https://github.com/organicmaps/organicmaps/issues/3881
-    // Set equal speeds here to avoid useless detours via service roads (Batumi_AvoidServiceDetour test).
-    {HighwayType::HighwayService, InOutCitySpeedKMpH(SpeedKMpH(10.0, 14.0))},
-    {HighwayType::HighwayResidential, InOutCitySpeedKMpH(SpeedKMpH(10.0, 14.0))},
-
-    {HighwayType::HighwayRoad, InOutCitySpeedKMpH(SpeedKMpH(10.0, 12.0))},
-    {HighwayType::HighwayTrack, InOutCitySpeedKMpH(SpeedKMpH(8.0, 12.0))},
-    {HighwayType::HighwayPath, InOutCitySpeedKMpH(SpeedKMpH(6.0, 12.0))},
-    {HighwayType::HighwayBridleway, InOutCitySpeedKMpH(SpeedKMpH(4.0, 12.0))},
-    {HighwayType::HighwayCycleway, InOutCitySpeedKMpH(SpeedKMpH(30.0, 20.0))},
-    {HighwayType::HighwayLivingStreet, InOutCitySpeedKMpH(SpeedKMpH(7.0, 8.0))},
+    // Note that roads with hwtag=yesbicycle get high speed of 0.9 * Cycleway.
+    /// @see Russia_UseTrunk test for Trunk weights.
+    {HighwayType::HighwayTrunk, InOutCitySpeedKMpH(SpeedKMpH(7.0, 17.0), SpeedKMpH(9.0, 19.0))},
+    {HighwayType::HighwayTrunkLink, InOutCitySpeedKMpH(SpeedKMpH(7.0, 17.0), SpeedKMpH(9.0, 19.0))},
+    {HighwayType::HighwayPrimary, InOutCitySpeedKMpH(SpeedKMpH(10.0, 17.0), SpeedKMpH(12.0, 19.0))},
+    {HighwayType::HighwayPrimaryLink, InOutCitySpeedKMpH(SpeedKMpH(10.0, 17.0), SpeedKMpH(12.0, 19.0))},
+    {HighwayType::HighwaySecondary, InOutCitySpeedKMpH(SpeedKMpH(13.0, 17.0), SpeedKMpH(15.0, 19.0))},
+    {HighwayType::HighwaySecondaryLink, InOutCitySpeedKMpH(SpeedKMpH(13.0, 17.0), SpeedKMpH(15.0, 19.0))},
+    {HighwayType::HighwayTertiary, InOutCitySpeedKMpH(SpeedKMpH(14.0, 17.0), SpeedKMpH(17.0, 19.0))},
+    {HighwayType::HighwayTertiaryLink, InOutCitySpeedKMpH(SpeedKMpH(14.0, 17.0), SpeedKMpH(17.0, 19.0))},
+    {HighwayType::HighwayUnclassified, InOutCitySpeedKMpH(SpeedKMpH(13.0, 17.0), SpeedKMpH(15.0, 19.0))},
+    {HighwayType::HighwayResidential, InOutCitySpeedKMpH(SpeedKMpH(12.0, 15.0), SpeedKMpH(14.0, 17.0))},
+    // Service roads are usually narrower and not meant for passing through.
```

Comment:
I would dispute "not meant for passing through". And even ones built for car traffic and not designed for passing through can be still very useful for cyclists as cycleway substitute due to typically lower traffic.

this is based on extensive experience from Poland where https://www.openstreetmap.org/directions?engine=fossgis_valhalla_bicycle&route=50.067527%2C20.003867%3B50.068148%2C20.004073 is normal and using service for bypasses - either intended by road design or not - is often a great idea

can list more examples

Reply:
Yeap my comment is not clear enough, so I've just removed it.

Actually I was quite surprised to find that service roads seems to be commonly used to map in countryside in Germany, see the Germany_UseServiceCountrysideRoads case.
I would have expected either unclassifieds or grade1 tracks...

Reply:
>  I would have expected either unclassifieds or grade1 tracks...

I expect that many are grade1 tracks tagged for renderer

Reply:
Actually this kind of `highway=service` use (rather long countryside connections which are not to e.g. a specific property) looks like mismapping according to https://wiki.openstreetmap.org/wiki/Tag:highway=service

Or are there local (German, Poland, etc.) conventions which support this kind of usage?

Reply:
> Actually this kind of highway=service use (rather long countryside connections which are not to e.g. a specific property) looks like mismapping according to https://wiki.openstreetmap.org/wiki/Tag:highway=service

yes, those are `highway=track` - no idea why Germans tagged it those way. If you point me to some I will open notes (or you can open notes yourself, I guess)

(Poland has misuse of `highway=track` for unpaved driveways but there is ongoing cleanup)

I can give examples of some `service` that are validly mapped and are useful shortcuts/bypasses if it would be useful.

Reply:
> 
> yes, those are `highway=track` - no idea why Germans tagged it those way. If you point me to some I will open notes (or you can open notes yourself, I guess)

Please check the `Germany_UseServiceCountrysideRoads` case, ref https://github.com/organicmaps/organicmaps/issues/6027

Maybe we should remove/disable this case until the tagging is fixed.

> 
> I can give examples of some `service` that are validly mapped and are useful shortcuts/bypasses if it would be useful.

Yeap it'd be helpful! Both shortcuts and detours via service.

Reply:
cycleway interruptor, but as low traffic service it works OK:
https://www.openstreetmap.org/directions?engine=graphhopper_bicycle&route=50.067485%2C20.003859%3B50.068198%2C20.004041 

shortcut unavailable to vehicles in general: https://www.openstreetmap.org/directions?engine=graphhopper_bicycle&route=50.067845%2C19.900210%3B50.068379%2C19.897437

avoids megatraffic, then continues as living street: https://www.openstreetmap.org/directions?engine=graphhopper_bicycle&route=50.098161%2C19.890951%3B50.101826%2C19.888762

should be enough, I guess? Can find more